lib/ansible/modules/cloud/azure/azure_rm_sqlserver.py

```diff
@@ -0,0 +1,321 @@
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            tags=dict(
+                type='dict',
+                required=False
+            ),
+            location=dict(
+                type='str',
+                required=False
+            ),
+            admin_username=dict(
+                type='str',
+                required=False
+            ),
+            admin_password=dict(
+                type='str',
+                no_log=True,
+                required=False
+            ),
+            version=dict(
+                type='str',
+                required=False
+            ),
+            identity=dict(
+                type='str',
+                required=False
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.parameters = dict()
+        self.parameters['identity'] = dict()
+
+        self.results = dict(changed=False, state=dict())
+        self.mgmt_client = None
+        self.state = None
+        self.to_do = Actions.NoAction
+
+        super(AzureRMServers, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                             supports_check_mode=True,
+                                             supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            elif key == "tags":
+                self.parameters["tags"] = kwargs[key]
+            elif key == "location":
+                self.parameters["location"] = kwargs[key]
+            elif key == "admin_username":
+                self.parameters["administrator_login"] = kwargs[key]
+            elif key == "admin_password":
+                self.parameters["administrator_login_password"] = kwargs[key]
+            elif key == "version":
+                self.parameters["version"] = kwargs[key]
+            elif key == "identity":
+                self.parameters['identity']["type"] = kwargs[key]
+
+        old_response = None
+        results = dict()
+
+        self.mgmt_client = self.get_mgmt_svc_client(SqlManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        try:
+            resource_group = self.get_resource_group(self.resource_group)
+        except CloudError:
+            self.fail('resource group {0} not found'.format(self.resource_group))
+
+        if not ("location" in self.parameters):
```

Comment:
This doesn't seem to be working, I got `The error was: msrest.exceptions.ValidationError: Parameter 'Server.location' can not be None` in an error which means this musn't work. Try doing;

```
if self.location is None:
    self.location = resource_group.location
```